ego/cli/install.go

```diff
@@ -0,0 +1,154 @@
+var ErrTargetNotSupported = errors.New("install target not supported")
+var ErrInstallationFail = errors.New("installation fail")
+var ErrExitCodeValue = errors.New("exit code not 0")
+var ErrSysInfoFail = errors.New("could not determine os details")
+
+type installInfoV1 struct {
+	Desc     map[string]string
+	Commands map[string]map[string][]string
+}
+
+type installInfo struct {
+	V1 installInfoV1 `json:"1.0"`
+}
+
+func (i installInfo) getDescriptions() map[string]string {
+	return i.V1.Desc
+}
+
+func (i installInfo) getCommands(osDetails string) map[string][]string {
+	return i.V1.Commands[osDetails]
+}
+
+// Determine relevant information about os and sgx compatibility. Then pass them with osDetails to install
+func (c *Cli) Install(ask func(string) bool, component string) error {
+
+	var si sysinfo.SysInfo
+	si.GetSysInfo()
+	vendor := si.OS.Vendor
+	version := si.OS.Version
+	if vendor == "" || version == "" {
+		fmt.Println("Could not determine necessary details about operating system. Exiting...")
+		return ErrSysInfoFail
+	}
+
+	var osDetails string
+	if !cpuid.CPU.Supports(cpuid.SGX) { // determine support for sgx
+		osDetails = strings.Join([]string{vendor, version, "nonsgx"}, "-")
+	} else {
+		if cpuid.CPU.Supports(cpuid.SGXLC) { // determine support for sgxflc
+			osDetails = strings.Join([]string{vendor, version, "flc"}, "-")
+		} else {
+			osDetails = strings.Join([]string{vendor, version, "nonflc"}, "-")
+		}
+	}
+	return c.install(ask, osDetails, component, "https://raw.githubusercontent.com/edgelesssys/ego/feat/install/src/install.json")
+}
+
+func (c *Cli) install(ask func(string) bool, osDetails string, component string, jsonURL string) error {
+	body, err := httpGet(jsonURL)
+	if err != nil {
+		return err
+	}
+
+	// Parse json data from url into installInfo structure
+	var installInfo installInfo
+	err = json.Unmarshal(body, &installInfo)
+	if err != nil {
+		return err
+	}
+
+	// Remove leading white spaces, required to be able to access infoInstall structure
+	component = strings.TrimSpace(component)
+
+	if component == "" || component == "desc" { // Print the description of the json entry in the latest version
+		descstring := "\n\tDescription of components\n"
+		for target, desc := range installInfo.getDescriptions() {
+			descstring = descstring + "\n" + strings.Join([]string{target, desc}, ": ")
+		}
+		fmt.Println(descstring)
+		return nil
+	} else {
+		osCommands := installInfo.getCommands(osDetails)
+		if osCommands == nil { // User system is not supported in the latest version. Returns error
+			fmt.Println("\nYour system is not supported in the latest version")
+			return ErrOSNotSupported
+		} else if osCommands[component] == nil { // The component to install is not supported in the latest version for the users system. Returns error
+			fmt.Println("\nYour system does not support this installation in the latest version")
+			return ErrTargetNotSupported
```

Comment:
I think we should treat this case as "component not found" and print the available components